plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -426,13 +454,18 @@ export class MongoDBInstrumentation extends InstrumentationBase<
     // mongodb is using "tick" when calling a callback, this way the context
     // in final callback (resultHandler) is lost
     const activeContext = context.active();
+    const instrumentation = this;
     return function patchedEnd(this: {}, ...args: unknown[]) {
       const error = args[0];
       if (error instanceof Error) {
         span.setStatus({
           code: SpanStatusCode.ERROR,
           message: error.message,
         });
+      } else {
+        const config = instrumentation.getConfig();
+        const results = args[1] as CommandResult;
+        instrumentation._handleExecutionResult(span, config, results);
```

Comment:
i think this should be called in all scenarios and not just for errors

Reply:
We are actually calling it only in successful cases - in case we have an error only the status of the span is being set.
Did you mean we have to add it for the error cases as well?

Reply:
Yeah sorry, i meant to say to add in every case

Reply:
That is a great idea! 
Right now we tried to be aligned with other response hook implementations on other instrumentations (such as in [Redis](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/93753ec88b228405ef5ce9a4d25500455766f5cc/plugins/node/opentelemetry-instrumentation-redis/src/utils.ts#L128) and ioredis)
Perhaps it should be done in a separate task, WDYT?

Reply:
I think it can be done here and the feature will be completed.